### ISSUE TITLE
beta17 batch normalization

### ISSUE THREAD

**Issue by zsslcma** [vlfeat/matconvnet#535]
hello everyone,
I'm using matconvnet_17 to train a network, which adds a bnorm after every conv layer.Now, I have a question that when testing an image, how to compute the mean and variance? Actually, I have seen the code which adds the mean and variance of every batch up, but the results is significantly different to that of every batch, I don't know why. In addition, can someone tell me why the code below can get the mean and variance of the training set? Thanks in advance!

![image](https://cloud.githubusercontent.com/assets/19218296/15064477/48c4a426-1387-11e6-9345-bc98e9e534e8.png)

**Comment by bazilas** [comment]
You don't compute them during testing. You provide them from your training.

Try to update to the latest version. This process already automatized there. Just don't forget to give the test flag, during test.

**Comment by zsslcma** [comment]
@bazilas , thank you very much
The version of matconvnet I'm using is beta17, where the implementation of calculating the mean and variance during training is latest, like the code in the red box of the picture, but I don't know why it works, I mean how the calculation method(the code in red box) can get the mean and variance which will be used during test.
Thank you again, bazilas
